lib/ansible/modules/package_facts.py

```diff
@@ -376,6 +376,31 @@ def get_package_details(self, package):
         return dict(zip(self.atoms, package.split()))
 
 
+class APK(CLIMgr):
+
+    CLI = 'apk'
+    atoms = ['name', 'version']
+
+    def list_installed(self):
+        rc, out, err = module.run_command([self._cli, 'info', '-v'])
+        if rc != 0 or err:
+            raise Exception("Unable to list packages rc=%s : %s" % (rc, err))
+        return out.splitlines()
+
+    def get_package_details(self, package):
+        raw_pkg_details = {}
+        for line in package.splitlines():
+            m = re.match(r"([\w ].*?)-([0-9-\.r]+)", line.decode("utf-8"))
```

Comment:
As reference, this regex doesn't work for these packages:

```
libcrypto1.1-1.1.1g-r0
libssl1.1-1.1.1g-r0
```

Those are just a few defaults that I tested against.

Additionally, this should use `to_native(line)` instead of `line.decode`

Reply:
addressed in latest